int16 deviceId
string name